checks.d/agent_metrics.py

```diff
@@ -0,0 +1,151 @@
+        Calls those methods and stores the raw output
+
+        :returns a dictionary of statistic_name: value
+        """
+        process_metrics = instance.get('process_metrics', self.init_config.get('process_metrics', None))
+        if not process_metrics:
+            self.log.error('No metrics configured for AgentMetrics check!')
+            return {}
+
+        methods, metric_types = zip(
+            *[(p['name'], p.get('type', 'gauge'))
+                for p in process_metrics if _is_affirmative(p.get('active'))]
+        )
+
+        names_to_metric_types = {}
+        for i, m in enumerate(methods):
+            names_to_metric_types[AgentCheck._get_statistic_name_from_method(m)] = metric_types[i]
+
+        stats = AgentCheck._collect_stats(methods, log=self.log)
+        return stats, names_to_metric_types
+
+    def _send_single_metric(self, metric_name, metric_value, metric_type):
+        if metric_type == 'gauge':
+            self.gauge(metric_name, metric_value)
+        elif metric_type == 'rate':
+            self.rate(metric_name, metric_value)
+        else:
+            raise UnsupportedMetricType(metric_name, metric_type)
+
+    def _register_psutil_metrics(self, stats, names_to_metric_types):
+        """
+        Saves sample metrics from psutil
+
+        :param stats: a dictionary that looks like:
+        {
+         'memory_info': OrderedDict([('rss', 24395776), ('vms', 144666624)]),
+         'io_counters': OrderedDict([('read_count', 4536),
+                                    ('write_count', 100),
+                                    ('read_bytes', 0),
+                                    ('write_bytes', 61440)])
+         ...
+         }
+
+         This creates a metric like `datadog.agent.collector.{key_1}.{key_2}` where key_1 is a top-level
+         key in `stats`, and key_2 is a nested key.
+         E.g. datadog.agent.collector.memory_info.rss
+        """
+
+        base_metric = 'datadog.agent.collector.{0}.{1}'
+        #TODO: May have to call self.normalize(metric_name) to get a compliant name
+        for k, v in stats.items():
+            metric_type = names_to_metric_types[k]
+            if isinstance(v, dict):
+                for _k, _v in v.items():
+                    full_metric_name = base_metric.format(k, _k)
+                    self._send_single_metric(full_metric_name, _v, metric_type)
+            else:
+                full_metric_name = 'datadog.agent.collector.{0}'.format(k)
+                self._send_single_metric(full_metric_name, v, metric_type)
+
+    def set_metric_context(self, payload, context):
+        self._collector_payload = payload
+        self._metric_context = context
+
+    def get_metric_context(self):
+        return self._collector_payload, self._metric_context
+
+    def check(self, instance):
+        in_developer_mode = self.agentConfig.get('developer_mode', False)
+        if in_developer_mode and psutil is not None:
```

Comment:
This is redundant with your 
```python
self.in_developer_mode
```
https://github.com/DataDog/dd-agent/blob/talwai/agent_dev_mode/checks/__init__.py#L303